src/usr/share/gpgmailer/gpgmailer.py

```diff
@@ -116,6 +120,45 @@ def start_monitoring(self):
                                   str(exception))
                 self.logger.error(traceback.format_exc())
 
+    def _read_and_send_message(self, file_name, loop_start_time):
+        """Attempt to build a message and send it while handling exceptions for individual
+        messages.
+
+        file_name: The name of the message file in the outbox directory. Not a full path.
+        loop_start_time: The time associated with the current program loop from which all PGP
+          key expiration checks are based.
+        """
+        try:
+            message_dict = self._read_message_file(file_name)
+
+            # Set default subject if the queued message does not have one.
+            if message_dict['subject'] is None:
+                message_dict['subject'] = self.config['default_subject']
+
+            encrypted_message = self._build_encrypted_message(
+                message_dict, loop_start_time)
+
+            self._send_mail(mime_message=encrypted_message,
+                            recipients=self.valid_recipient_emails)
+            self.logger.info('Message %s sent successfully.', file_name)
+
+            os.remove(os.path.join(self.outbox_path, file_name))
+
+        except gpgkeyverifier.NoUsableKeysException as no_usable_keys:
+            # This exception will prevent all messages from sending, so re-raise it to
+            #   restart the loop.
+            raise no_usable_keys
+
+        except gpgkeyverifier.SenderKeyExpiredException as sender_key_expired:
+            # This exception will prevent all messages from sending, so re-raise it to
+            #   restart the loop.
```

Comment:
I'd say "This exception should abort the program, so we just re-raise it." Same for the above message.